agent/grpc-external/services/peerstream/replication.go

```diff
@@ -219,14 +223,29 @@ func (s *Server) handleUpdateService(
 	peerName string,
 	partition string,
 	sn structs.ServiceName,
+	mutableStatus *MutableStatus,
 	pbNodes *pbservice.IndexedCheckServiceNodes,
+	logger hclog.Logger,
```

Comment:
🤔 I wonder if the logger for this stream should be stored in the stream tracker. This is fine for now, but I will think on this some more and maybe write up some future tasks to expand the role of the MutableStatus.